### PR TITLE
H2STORE test cases

### PR DESCRIPTION
**Pull request by svenn-t**
Depends on https://github.com/OPM/opm-common/pull/3445 and https://github.com/OPM/opm-simulators/pull/4528 
These test cases are the same as for CO2STORE, but with H2STORE keyword enabled.

### CODE REVIEW

File: H2STORE_GW.DATA [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+
+-- Copyright (C) 2021 NORCE
+
+-- Test case for gas + water + disgasw (dissolved gas in water phase)
+
+----------------------------------------------------------------------------
+RUNSPEC
+----------------------------------------------------------------------------
+TITLE
+   CO2STORE GAS-WATER DISGASW
```
Comment (goncalvesmachadoc) [inline H2STORE_GW.DATA:14]: CO2STORE -> H2STORE

File: H2STORE_GW.DATA [hunk 2]
```diff
@@ -0,0 +1,186 @@
+-- This reservoir simulation deck is made available under the Open Database
+-- License: http://opendatacommons.org/licenses/odbl/1.0/. Any rights in
+-- individual contents of the database are licensed under the Database Contents
+-- License: http://opendatacommons.org/licenses/dbcl/1.0/
+
+-- Copyright (C) 2021 NORCE
```
Comment (goncalvesmachadoc) [inline H2STORE_GW.DATA:6]: 2021 -> 2023

File: H2STORE_GW.DATA [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+0	0	0.0
+1	1 	0.0 /
+
+SWFN
+0	0	0.0
+1	1 	0.0 /
+
+SALINITY
+ 0.7/ 35-40g/l  -> 35-40g /kg -> 0.63-0.72 mol/g
+
```
Comment (goncalvesmachadoc) [inline H2STORE_GW.DATA:93]: 0.7/ 35-40g/l  -> 35-40g /kg -> 0.63-0.72 mol/g
->
0.7/ -- 35-40g/l  -> 35-40g /kg -> 0.63-0.72 mol/g

File: H2STORE_GW.DATA
Comment (bska) [inline H2STORE_GW.DATA:1]: This file (`H2STORE_GW.DATA`) appears to have been added **outside** of the new `h2store` directory&mdash;i.e., in the top-level `opm-tests` directory.  Is that intentional?  If not I'd prefer if it were moved into the `h2store` directory.

File: H2STORE_GW.DATA (hunk 1 above)
Comment (svenn-t) [reply H2STORE_GW.DATA]: Corrected, thanks!

File: H2STORE_GW.DATA (hunk 2 above)
Comment (svenn-t) [reply H2STORE_GW.DATA]: Corrected, thanks!

File: H2STORE_GW.DATA (hunk 3 above)
Comment (svenn-t) [reply H2STORE_GW.DATA]: Corrected, thanks!

File: H2STORE_GW.DATA
Comment (svenn-t) [reply H2STORE_GW.DATA:1]: Sorry, that was not intentional. Moved it inside h2store directory now.

**Comment by totto82** [comment]
@bska and @goncalvesmachadoc Can this be merged now? I would like to get this in before merging the update on the H2STORE option to make sure we have proper tests for the new functionality.

**Review by bska — COMMENTED** [review]
> Can this be merged now?

I don't see why not as most of this is straightforward.  I'll let @goncalvesmachadoc have the final say, though, because I don't know enough about the physics here to know if I'm missing some subtle detail.

File: h2store/H2STORE.DATA [hunk 4]
```diff
@@ -0,0 +1,150 @@
+-- This reservoir simulation deck is made available under the Open Database
+-- License: http://opendatacommons.org/licenses/odbl/1.0/. Any rights in
+-- individual contents of the database are licensed under the Database Contents
+-- License: http://opendatacommons.org/licenses/dbcl/1.0/
+
+-- Copyright (C) 2023 NORCE
+
```
Comment (goncalvesmachadoc) [inline h2store/H2STORE.DATA:7]: Please add short description of the case.

File: h2store/H2STORE.DATA [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+   1 10 1 1 /
+
+UNIFIN
+UNIFOUT
+
+----------------------------------------------------------------------------
+GRID
+----------------------------------------------------------------------------
+
+NOECHO
```
Comment (goncalvesmachadoc) [inline h2store/H2STORE.DATA:42]: Remove NOECHO.

File: h2store/H2STORE_DIFFUSIVE.DATA [hunk 6]
```diff
@@ -0,0 +1,151 @@
+-- This reservoir simulation deck is made available under the Open Database
+-- License: http://opendatacommons.org/licenses/odbl/1.0/. Any rights in
+-- individual contents of the database are licensed under the Database Contents
+-- License: http://opendatacommons.org/licenses/dbcl/1.0/
+
+-- Copyright (C) 2023 NORCE
+
```
Comment (goncalvesmachadoc) [inline h2store/H2STORE_DIFFUSIVE.DATA:7]: Please add short description of the case.

File: h2store/H2STORE_DIFFUSIVE.DATA [hunk 7]
```diff
@@ -0,0 +1,151 @@
...
+   1 10 1 1 /
+
+UNIFIN
+UNIFOUT
+
+----------------------------------------------------------------------------
+GRID
+----------------------------------------------------------------------------
+
+NOECHO
```
Comment (goncalvesmachadoc) [inline h2store/H2STORE_DIFFUSIVE.DATA:43]: Remove NOECHO.

File: h2store/H2STORE_DISGAS.DATA [hunk 8]
```diff
@@ -0,0 +1,151 @@
+-- This reservoir simulation deck is made available under the Open Database
+-- License: http://opendatacommons.org/licenses/odbl/1.0/. Any rights in
+-- individual contents of the database are licensed under the Database Contents
+-- License: http://opendatacommons.org/licenses/dbcl/1.0/
+
+-- Copyright (C) 2023 NORCE
+
```
Comment (goncalvesmachadoc) [inline h2store/H2STORE_DISGAS.DATA:7]: Please add short description of the case.

File: h2store/H2STORE_DISGAS.DATA [hunk 9]
```diff
@@ -0,0 +1,151 @@
...
+   1 10 1 1 /
+
+UNIFIN
+UNIFOUT
+
+----------------------------------------------------------------------------
+GRID
+----------------------------------------------------------------------------
+
+NOECHO
```
Comment (goncalvesmachadoc) [inline h2store/H2STORE_DISGAS.DATA:43]: Remove NOECHO.

File: h2store/H2STORE_ENERGY.DATA [hunk 10]
```diff
@@ -0,0 +1,164 @@
+-- This reservoir simulation deck is made available under the Open Database
+-- License: http://opendatacommons.org/licenses/odbl/1.0/. Any rights in
+-- individual contents of the database are licensed under the Database Contents
+-- License: http://opendatacommons.org/licenses/dbcl/1.0/
+
+-- Copyright (C) 2023 NORCE
+
```
Comment (goncalvesmachadoc) [inline h2store/H2STORE_ENERGY.DATA:7]: Please add short description of the case.

File: h2store/H2STORE_ENERGY.DATA [hunk 11]
```diff
@@ -0,0 +1,164 @@
...
+   1 10 1 1 /
+
+UNIFIN
+UNIFOUT
+
+----------------------------------------------------------------------------
+GRID
+----------------------------------------------------------------------------
+
+NOECHO
```
Comment (goncalvesmachadoc) [inline h2store/H2STORE_ENERGY.DATA:45]: Remove NOECHO.

File: h2store/H2STORE_GW.DATA [hunk 12]
```diff
@@ -0,0 +1,186 @@
...
+   2 10 1 2 /
+
+UNIFIN
+UNIFOUT
+
+----------------------------------------------------------------------------
+GRID
+----------------------------------------------------------------------------
+
+NOECHO
```
Comment (goncalvesmachadoc) [inline h2store/H2STORE_GW.DATA:49]: Remove NOECHO.

File: h2store/H2STORE_GW.DATA [hunk 13]
```diff
@@ -0,0 +1,186 @@
...
+/
+
+WCONPROD
+-- Item #:1	2      3     4	   5     9
+	'PROD' 'OPEN' 'BHP' 1*     1* 3* 190 /
+/
+
+-- 30 day
+TSTEP
+100*1
```
Comment (goncalvesmachadoc) [inline h2store/H2STORE_GW.DATA:180]: Is it necessary to have a longer run for this case in particular? All other cases are 30*1.

**Review by goncalvesmachadoc — CHANGES_REQUESTED** [review]

**Pull request merged**